### PR TITLE
BL-1066-be-fe-authentication-and-authorization-hook-ups

### PR DESCRIPTION
**Pull request by mina-mikhael**
## Ticket BL-1066-be-fe-authentication-and-authorization-hook-ups

Intitially we had a BE middle-ware that was built to receive id_tokens, We needed to change it to receive an `accessToken` instead. 

The FE needed to get that token on demand and make sure that it's set on the `authorization header` on the request. That token was sometimes undefined.

Ticket objective is to wire FE API calls to BE private Endpoints through exchanging and verifying access_tokens by the auth. provider (auth0).

#### Back-End:

- Changed the audience in the .env file in order to verify access tokens instead of id tokens

#### Front-End:

- Added the audience in the .env file in order to add the add the correct header to the access token (jwt), to be verified later by the BE and the authProvider.
- Added the audience prop to the auth0Provider.
- Refactored the useAxiosWithAuth0 custom hook to create a new axios instance with the access token attached to the headers whenever we have authenticated user logged in.
- Fixed a bug in all private routes components that is supposed to consume the useAxiosWithAuth0 custom hook to add the access token to the request header when making api-call to private BE End-point.

### Video Link

[Loom Video](https://www.loom.com/share/9e12323c71734982b056eb55159f8f56)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

File: src/components/common/MemosTable/AddReply/showReply.js [hunk 1]
```diff
@@ -5,16 +5,14 @@ import moment from 'moment';
 
 function ShowReply(props) {
   const { note_id } = props;
-  const { axiosWithAuth } = useAxiosWithAuth0();
+  const axiosWithAuth = useAxiosWithAuth0();
```
Comment (ryan-hamblin) [inline src/components/common/MemosTable/AddReply/showReply.js:8]: in the future, when you come across a linter warning go ahead and update in the file you're working on.

File: src/hooks/useAxiosWithAuth0.js [hunk 2]
```diff
@@ -2,50 +2,29 @@ import { useEffect } from 'react';
...
 import { useAuth0 } from '@auth0/auth0-react';
 
-/*
-NOTE:
-React requires that we can only call hooks in function components or within other hooks. As axiosWithAuth is simply a helper function we've built a useAxiosWithAuth0() hook in which we've called both axiosWithAuth AND useAuth0().
-axiosWithAuth serves the same function as it typically does. useAuth0() gives access to getAccessTokenSilently() for authentication/authorization purposes. 
-This hook below is exported and used throughout the app in place of axiosWithAuth.
-THIS useAxiosWithAuth0 HOOK SIMPLY SERVES TO HELP CONNECT THE AXIOS CALL AND THE AUTHENTICATION.
-*/
+const axios_instance = axios.create({
```
Comment (ryan-hamblin) [inline src/hooks/useAxiosWithAuth0.js:5]: Awesome! A little global to set the axios.create fn in scope! This is how it should've been done.

File: src/hooks/useAxiosWithAuth0.js [hunk 3]
```diff
@@ -2,50 +2,29 @@ import { useEffect } from 'react';
...
-      '/',
-      '/apply',
-      '/apply/mentee',
-      '/apply/mentor',
-      '/apply/success',
-    ];
-
-    /********************************** */
+    if (isAuthenticated) {
+      axios_instance.interceptors.request.use(
```
Comment (ryan-hamblin) [inline src/hooks/useAxiosWithAuth0.js:14]: can you cross link to the axios docs in a code comment somewhere here in the file? This is brilliant! https://axios-http.com/docs/interceptors

After talking through this logic surrounding `isAuthenticated` and learning a bit about axios.interceptors I feel confident this is the closest to the final solution we'd like too see!

**Review by ryan-hamblin — CHANGES_REQUESTED** [review]
The only change that I request will be in the documentation of the `interceptors` api on Axios.

**Review by TylrPopcorn — APPROVED** [review]
I think this should move the project toward a positive direction

**Review by JamesFincher — APPROVED** [review]
Clean. Will def help push things forward. Great work!

**Pull request merged**